### PR TITLE
✨  [Story auto-analytics] Added validation and tests

### PR DESCRIPTION
**Pull request by mszylkowski**
Added validation for amp-story-auto-analytics

### CODE REVIEW

File: extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.out [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+|    <amp-story standalone title="My Story" publisher="Me" publisher-logo-src="http://me.com/logo.png" poster-portrait-src="http://me.com/poster.jpg">
+|      <amp-story-auto-analytics gtag-id="ANALYTICS-ID"></amp-story-auto-analytics>
+|      <amp-story-page id="cover" auto-advance-after="1s">
+|        <amp-story-grid-layer template="fill">
+|          <h1>Title</h1>
+|        </amp-story-grid-layer>
+|      </amp-story-page>
+|    </amp-story>
+|  </body>
+|  </html>
```
Comment (Enriqe) [inline extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.out:43]:
```suggestion
|  </html>

```

**Review by Enriqe — APPROVED** [review]

File: extensions/amp-story-auto-analytics/validator-amp-story-auto-analytics.protoascii [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-story-auto-analytics>
+  html_format: AMP
+  tag_name: "AMP-STORY-AUTO-ANALYTICS"
+  requires_extension: "amp-story-auto-analytics"
+  attrs: {
```
Comment (gmajoulet) [inline extensions/amp-story-auto-analytics/validator-amp-story-auto-analytics.protoascii:31]: This one should be mandatory.

Can we validate it using a regex like `UA-[0-9]+-[0-9]+`?

File: extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.out (hunk 1 above)
Comment (mszylkowski) [reply extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.out:43]: The default output of validation from `gulp validator --update_tests` doesn't come with the newline 😢

File: extensions/amp-story-auto-analytics/validator-amp-story-auto-analytics.protoascii (hunk 2 above)
Comment (mszylkowski) [reply extensions/amp-story-auto-analytics/validator-amp-story-auto-analytics.protoascii:31]: Added mandatory, but as we discussed regex would be complex due to the number of different types of tags so we're not enforcing a regex on the value (+ future compatibility)

File: extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<!--
```
Comment (honeybadgerdontcare) [inline extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.html:1]: Could you add a comment above the line that is being validated with something like `<!-- Valid: has correct gtag attribute and value -->`.

**Review by honeybadgerdontcare — APPROVED** [review]
Pending already existing feedback (mandatory, value_regex) the validation changes look fine. Please only submit after addressing those.

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.out (hunk 1 above)
Comment (Enriqe) [reply extensions/amp-story-auto-analytics/0.1/test/validator-amp-story-auto-analytics.out:43]: Yep, but you can still add it manually for now tho

/cc @ampproject/wg-caching

File: extensions/amp-story-auto-analytics/validator-amp-story-auto-analytics.protoascii (hunk 2 above)
Comment (honeybadgerdontcare) [reply extensions/amp-story-auto-analytics/validator-amp-story-auto-analytics.protoascii:31]: Could you link the discussion?

Comment (honeybadgerdontcare) [reply extensions/amp-story-auto-analytics/validator-amp-story-auto-analytics.protoascii:31]: And regarding `future compatibility`. It's easier to be more restrictive and ease those than to be less restrictive and try to tighten it.

**Review by honeybadgerdontcare — APPROVED** [review]

**Pull request merged**